packages/datagrid/src/components/HeaderCellRenderer/HeaderCellRenderer.component.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { IHeaderParams } from 'ag-grid-community';
+import classNames from 'classnames';
+import truncate from 'lodash/truncate';
+
+import { Icon, Tooltip, ButtonIcon, StackHorizontal } from '@talend/design-system';
+import { QualityBar } from '@talend/react-components';
+
+import { QUALITY_INVALID_KEY, QUALITY_EMPTY_KEY, QUALITY_VALID_KEY } from '../../constants';
+import { HeaderComponentParams } from '../../types';
+
+import theme from './HeaderCellRenderer.scss';
+
+export type HeaderRendererProps = IHeaderParams &
+	HeaderComponentParams & {
+		onFocusedColumn(colId: string): void;
+		onKeyDown(event: React.KeyboardEvent<HTMLElement>, colId: string): void;
+	};
+
+/**
+ * Props are provided by column definition headerComponentParams
+ * https://www.ag-grid.com/javascript-data-grid/component-header/#custom-header-parameters
+ */
+export default function HeaderCellRenderer({
+	column,
+	quality,
+	typeLabel,
+	semanticTypeLabel,
+	required,
+	description,
+	isLoading,
+	draftType,
+	menuProps,
+	qualityBarProps,
+	displayName,
+	onFocusedColumn,
+	onKeyDown,
+}: HeaderRendererProps): JSX.Element {
+	const { t } = useTranslation();
+	const colId = column.getColId();
+
+	return (
+		<div
+			className={classNames(theme['header-cell'], {
+				[theme['header-cell--loading']]: isLoading,
+				[theme['header-cell--type-draft']]: !!draftType,
+			})}
+		>
+			<div
+				className={theme['header-cell__button']}
+				onClick={() => onFocusedColumn(colId)}
```

Comment:
Does the focus only happen on the "click" event? Maybe it can be done with tab navigation?

Reply:
tab navigation is already broken... I'll have to work on it for the multi selection feature anyway